indico/modules/events/api.py

```diff
@@ -551,9 +551,9 @@ def _build_event_api_data(self, event):
             'modificationDate': self._serialize_date(event.as_legacy.getModificationDate()),
             'creationDate': self._serialize_date(event.created_dt),
             'creator': self._serialize_person(event.creator, person_type='Avatar', can_manage=can_manage),
-            'hasAnyProtection': event.as_legacy.hasAnyProtection(),
+            'hasAnyProtection': event.effective_protection_mode != ProtectionMode.public,
             'roomMapURL': event.room.map_url if event.room else None,
-            'visibility': Conversion.visibility(event.as_legacy),
+            'visibility': event.visibility,
```

Comment:
Look at `Conversion.visibility`, it already contains it! I was talking about the other one that has `{name: ..}` only atm